src/bin/cargo/main.rs

```diff
@@ -1,6 +1,4 @@
-#![warn(rust_2018_idioms)] // while we're getting used to 2018
-#![allow(clippy::all)]
-#![warn(clippy::disallowed_methods)]
+#![allow(clippy::self_named_module_files)] // false positive in `commands/build.rs`
```

Comment:
Could we narrow this to just where its needed?

Reply:
I've tried, but in vain.

BTW, here is a report of the false-positive: https://github.com/rust-lang/rust-clippy/issues/9602